pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -372,5 +399,46 @@ private static void getAggregations(FilterContext filter, List<FunctionContext>
         getAggregations(filter.getPredicate().getLhs(), aggregations);
       }
     }
+
+    /**
+     * Helper method to extract the columns (IDENTIFIER expressions) for the query.
+     */
+    public void extractColumns(QueryContext query) {
```

Comment:
why is this a public method? who is invoking this and from where

Reply:
Good catch, changed to private